htmresearch/encoders/cio_encoder.py

```diff
@@ -167,8 +167,10 @@ def getUnionEncoding(self, text):
 
   def getWindowEncoding(self, tokens, minSparsity=0.0):
     """
-    The encoding representation of a given token is a union of its bitmap with
-    the immediately previous tokens' bitmaps, up to the maximum sparsity.
+    The encodings simulate a "sliding window", where the encoding representation
```

Comment:
@BoltzmannBrain This function is confusing to me, particularly why it is doing what it is doing (it is *not* a sliding window), how/why a client would actually use this, and why we have so many separate encoding functions.  I don't mind if you merge this, but we need to discuss the design of the whole encoder.